whatrecord/parse.py

```diff
@@ -74,7 +75,20 @@ def parse(
     """
     standin_directories = standin_directories or {}
 
-    filename = pathlib.Path(filename)
+    if str(filename) == "-":
+        contents = sys.stdin.read()
+        filename = pathlib.Path("/dev/stdin")
+
+        if format is None:
+            options = [fmt.name for fmt in list(FileFormat)]
+            raise ValueError(
+                f"Format must be specified when piping data from standard "
+                f"input.  Choose one of: {options}"
```

Comment:
Maybe this check should come before you read from stdin

Reply:
Good call; addressed in d843fd5